src/auth.ts

```diff
@@ -0,0 +1,81 @@
+/*
+auth.ts - Provides auth related methods for the Bluebutton class
+*/
+import crypto from "crypto";
+
+import BlueButton from ".";
+import { authData, tokenPostData } from "./types/auth";
+
+function base64URLEncode(buffer: Buffer): string {
+  return buffer
+    .toString("base64")
+    .replace(/\+/g, "-")
+    .replace(/\//g, "_")
+    .replace(/=/g, "");
+}
+
+function sha256(str: string): Buffer {
+  return crypto.createHash("sha256").update(str).digest();
+}
+
+type pkceData = {
```

Comment:
```suggestion
type PkceData = {
```

Generally types/classes are capitalized

Reply:
@nbragdon Updated this and other related type/class vars. Will do another pass to see if there are more naming changes. TY!